src/MSBuild/LiveLogger/MessageNode.cs

```diff
@@ -69,7 +69,7 @@ public string ToANSIString()
                         ANSIBuilder.Formatting.ForegroundColor.Red)}";
                 case MessageType.HighPriorityMessage:
                 default:
-                    return $"ℹ️ Message {Code}: {ANSIBuilder.Formatting.Italic(Message)}";
+                    return $"ℹ️ {(Code != null ? Code + ": " : string.Empty)} {ANSIBuilder.Formatting.Italic(Message)}";
```

Comment:
This requires concatenating a string and breaks out of some optimizations (on .NET Core). If you write it a bit oddly you can get them back though:

```suggestion
                    return $"ℹ️ {Code}{(Code is not null ? ": " : string.Empty)} {ANSIBuilder.Formatting.Italic(Message)}";
```

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+TgAxUQAmMA3PYxi6QCyMAM4iAhgHNuvBiSatKEAHZhxAFxEAKAJS1ZjOQHZSAEgBEgTkJA8H+0tSzqQCEAXlLKArgBsvpAPxsjgDUpOYgoaTh/MwAdACiALYADuoAnjoAvrTCYlIwGeY8BIYZBgby0aQA6jAAllCcUABCHuoAMqIiAII+EGC6+sWGfCYWNrQO+TT2HDCktSLuEOru3r4BYRFRLPHJaZnZnXkFRSWEGUA== shows the compiled differences.

Reply:
That looks smart (and also cleaner), thx!